File: distance_calc.lua
function distance2d(from, to)
    diffX = math.abs(from[1] - to[1])
    diffY = math.abs(from[2] - to[2])
    delta = math.sqrt((diffX ^ 2) + (diffY ^ 2))
    return delta
end

function distance3d(from, to)
    from2d = {from[1], from[3]}
    to2d = {to[1], to[3]}
    diffX = distance2d(from2d, to2d)
    diffY = math.abs(from[2] - to[2])
    delta = math.sqrt((diffX ^ 2) + (diffY ^ 2))
    return delta
end

function main()
    a = {1.0, 2.0, 3.0}
    b = {4.0, 5.0, 6.0}
    print(distance3d(a, b))
end

main()

File: distance_calc.rb
def distance2d(from, to)
	diff_x = (from[0] - to[0]).abs
	diff_y = (from[1] - to[1]).abs
	delta = Math.sqrt((diff_x ** 2) + (diff_y ** 2))
	return delta
end

def distance3d(from, to)
	from2d = [from[0], from[2]]
	to2d = [to[0], to[2]]
	diff_x = distance2d(from2d, to2d)
	diff_y = (from[1] - to[1]).abs
	delta = Math.sqrt((diff_x ** 2) + (diff_y ** 2))
	return delta
end

if __FILE__ == $0
	a = [1, 2, 3]
	b = [4, 5, 6]
	puts(distance3d(a, b))
end